src/core/algorithms/fd/fd_algorithm.cpp

```diff
@@ -27,6 +27,20 @@ void FDAlgorithm::ResetState() {
     ResetStateFd();
 }
 
+std::list<FD>& FDAlgorithm::SortedFdList() {
+    fd_collection_.AsList().sort([](const FD& l_fd, const FD& r_fd) {
+        if (l_fd.GetLhs().GetArity() != r_fd.GetLhs().GetArity()) {
+            return l_fd.GetLhs().GetArity() < r_fd.GetLhs().GetArity();
+        }
+        if (l_fd.GetLhs() != r_fd.GetLhs()) {
+            return l_fd.GetLhs() < r_fd.GetLhs();
+        }
+        return l_fd.GetRhsIndex() < r_fd.GetRhsIndex();
+    });
+
+    return fd_collection_.AsList();
```

Comment:
nit:
I would use a variable for `fd_collection_.AsList()` to avoid repeating it. Also it will be a bit easier to read, since it'll be obvious that it's a lvalue ref
```cpp
std::list<FD>& fd_collection = fd_collection_.AsList();
fd_collection.sort();
return fd_collection;
```